src/robot_interface/robot_interface.py

```diff
@@ -90,3 +92,37 @@ def get_inspections(self, step: InspectionStep) -> Sequence[Inspection]:
 
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def get_pose(self) -> Pose:
+        """Return the pose of the robot.
+
+        Returns
+        -------
+        Pose
+            An Alitra pose containing a position, an orientation and a frame.
+
+        Raises
+        ------
+        RobotException
+            If the pose can't be retrieved.
+
+        """
+        raise NotImplementedError
```

Comment:
Does this scale?

Pose and battery status is not the only timeseries available from the robots and there will be different timeseries for different robots. Would we need to make individual interface functions for each type of data we wish to stream?

Reply:
Agree